client/src/reducers/controls.js

```diff
@@ -101,6 +139,39 @@ const Controls = (
       const categoricalAsBooleansMap = createCategoricalAsBooleansMap(world);
       const crossfilter = Crossfilter(world.obsAnnotations);
       const dimensionMap = World.createObsDimensionMap(crossfilter, world);
+
+      let worldVarDataCache = world.varDataCache;
+      /* var dimensions */
+
+      if (userDefinedGenes.length > 0) {
+        /*
+          verbose & slightly confusing that we also access this as an object
+          in controls rather than an array, should be abstracted into
+          util ie., createDimensionsFromBothListsOfGenes(userGenes, diffExp)
+        */
+        _.forEach(userDefinedGenes, (gene, index) => {
+          dimensionMap[gene] = World.createVarDimension(
+            /* "__var__" + */
+            world,
+            worldVarDataCache,
+            crossfilter,
+            gene
+          );
+        });
+      }
+
+      if (diffexpGenes.length > 0) {
```

Comment:
see above note about leaking / clobbering dimensions in the case where you have the same gene name in both userDefinedGenes and diffexpGenes.